Configure Path where soundfiles are dumped to

elektm93 opened the pull request:
Dear Alfred,

I had the problem to localize my snd-dumps, when running baresip in deamon mode. Due to lack of permissions, the got not created or dropped somewhere I couldn't find.

## Solution
- path where the soundfiles are dropped is now configurable (like in other modules)

The way how the destination-filename is constructed may not be an advanced one. Feel free to edit!

Best Regards,
Martin

elektm93 commented:
sorry - there was a segmentation fault in there. fixed it.

alfredh commented:
many thanks for your patch.

I get some errors when compiling (OSX and clang):

```
  CC [M]  build-x86_64/modules/sdl2/sdl.o
  LD [M]  sdl2.so
  CC [M]  build-x86_64/modules/sndfile/sndfile.o
modules/sndfile/sndfile.c:36:2: warning: excess elements in scalar initializer
        PL("./")
        ^~~~~~~~
../re/include/re_fmt.h:22:21: note: expanded from macro 'PL'
#define PL(s) {(s), sizeof((s))-1}
                    ^~~~~~~~~~~~~
```

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -13,6 +13,12 @@
+ * Example Configuration:
+ \verbatim
+  snd_path 					/tmp/
+	snd_file_prefix		dump
```
snd_file_prefix is not used ?

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -168,6 +184,13 @@ static struct aufilt sndfile = {
 {
 	aufilt_register(&sndfile);
+
+	if (conf_get(conf_cur(), "snd_path", &file_path)) {
```
as a general rule in baresip/libre, when using the `struct pl` type you should use the value
in the same function, because the data might be gone when the function returns.

please use a string instead and the function `conf_get_str` to read the config. The return value
can be ignored.

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -26,6 +32,10 @@ struct sndfile_dec {
+static struct pl file_path = {
+	PL("./")
+};
+
```
please use a string here (static char file_path[256] = ".")

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -72,6 +83,11 @@ static SNDFILE *openfile(const struct aufilt_prm *prm, bool enc)
+	strcat(filepath, "/");
+	strcat(filepath, filename);
+	strcpy(filename, filepath);
+
```
I think it is better to modify the code that creates the path instead, and always insert
the path in the beginning. for example:

```
	r = re_snprintf(filename, sizeof(filename),
			  "%s/dump-%H-%s.wav",
			  path,
			  timestamp_print, tm, enc ? "enc" : "dec");
	if (r <= 0) return NULL;
```

elektm93 commented:
Hi Alfred,

thank you for you comments! I've implemented everything as suggested.

* removed the commend for snd_file_prefix. I intended to implement further ideas, but I reassessed that thought after writing that comment.
* changed the static struct pl file_path to an string, hence also the compiling error for OSX and clang should be gone.
* the approach of of creating the filename is implemented as suggested

Sorry for the inconvinience and thank you for the creation of this awesome project!! 

Best Regards,
Martin

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -63,13 +70,15 @@ static void dec_destructor(void *arg)
 static SNDFILE *openfile(const struct aufilt_prm *prm, bool enc)
 {
 	char filename[128];
+	char filepath[256];
```
this variable is unused and can be removed.

alfredh commented on modules/sndfile/sndfile.c:
```diff
@@ -168,6 +177,11 @@ static struct aufilt sndfile = {
 {
 	aufilt_register(&sndfile);
+
+	conf_get_str(conf_cur(), "snd_path", &file_path, 256);
```
the last two parameters here should be:

`, file_path, sizeof(file_path)`

alfredh commented:
I am still getting some compile warning on OSX:

```
 make sndfile.so
  CC [M]  build-x86_64/modules/sndfile/sndfile.o
modules/sndfile/sndfile.c:73:7: warning: unused variable 'filepath' [-Wunused-variable]
        char filepath[256];
             ^
modules/sndfile/sndfile.c:181:39: warning: incompatible pointer types passing 'char (*)[256]' to parameter of type 'char *' [-Wincompatible-pointer-types]
        conf_get_str(conf_cur(), "snd_path", &file_path, 256);
                                             ^~~~~~~~~~
../re/include/re_conf.h:15:67: note: passing argument to parameter 'str' here
int conf_get_str(const struct conf *conf, const char *name, char *str,
                                                                  ^
2 warnings generated.
  LD [M]  sndfile.so
```

alfredh commented:
fixed in https://github.com/alfredh/baresip/commit/903ce9869c933e4f13995cd9f10c1498665cfd58
